apps/web/src/app/[locale]/challenge/[slug]/_context.tsx

```diff
@@ -0,0 +1,29 @@
+'use client';
```

Comment:
praise: this file looks great!

(non-blocking) thought: I wonder if there is a better name for this file, since `context` is an implementation detail for `useTrackVisibility`. Instead i might name this file with `hooks` in the name, or, `use-track-visibility.tsx` as it is tied to a single hook.

Reply:
yeah making this change